lib/msf/base/sessions/mainframe_shell.rb

```diff
@@ -73,37 +72,36 @@ def shell_read(length=-1, timeout=1)
   #
   def shell_write(buf)
     #mfimpl
-    return if not buf
+    return unless buf
 
     begin
       framework.events.on_session_command(self, buf.strip)
       rstream.write(Rex::Text.to_ibm1047(buf))
     rescue ::Rex::SocketError, ::EOFError, ::IOError, ::Errno::EPIPE => e
-      #print_error("Socket error: #{e.class}: #{e}")
       shell_close
       raise e
     end
   end
 
-  #
   def execute_file(full_path, args)
     #mfimpl
     raise NotImplementedError
   end
 
-  #
   def process_autoruns(datastore)
```

Comment:
Ah, okay, that's why you left out the ```raise```. Oops! I'll remove it for now, you can merge this PR, and then you can add the ```undef_method```?